client/components/publish-confirmation.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <v-layout row>
+    <v-container fill-height>
+      <v-layout fill-height>
+        <v-flex>
+          <v-card>
+            <v-card-title primary-title>
+              <div>
+                <h3>Do you wish to publish this document now?</h3>
+                <div>Once it's published it will not be editable anymore.</div>
+              </div>
+            </v-card-title>
+            <v-card-actions>
+              <v-btn flat color="primary" class="mx-0" @click="onCancelClick">Cancel</v-btn>
+              <v-btn flat color="primary" class="mx-0" @click="onPublishClick">Publish</v-btn>
```

Comment:
Disable the form while Meteor method call is being processed. See `user-signin` for an example.

Moreover, this call should probably happen only on the server, not on the client. So no simulation here. User does not expect latency compensation.